pkg/services/publicdashboards/service/service.go

```diff
@@ -167,23 +172,31 @@ func (pd *PublicDashboardServiceImpl) updatePublicDashboardConfig(ctx context.Co
 
 // BuildPublicDashboardMetricRequest merges public dashboard parameters with
 // dashboard and returns a metrics request to be sent to query backend
-func (pd *PublicDashboardServiceImpl) BuildPublicDashboardMetricRequest(ctx context.Context, dashboard *models.Dashboard, publicDashboard *PublicDashboard, panelId int64) (dtos.MetricRequest, error) {
+func (pd *PublicDashboardServiceImpl) BuildPublicDashboardMetricRequest(ctx context.Context, dashboard *models.Dashboard, publicDashboard *PublicDashboard, panelId int64, reqDTO *PublicDashboardQueryDTO) (dtos.MetricRequest, error) {
 	if !publicDashboard.IsEnabled {
 		return dtos.MetricRequest{}, ErrPublicDashboardNotFound
 	}
 
 	queriesByPanel := models.GroupQueriesByPanelId(dashboard.Data)
 
-	if _, ok := queriesByPanel[panelId]; !ok {
+	queries, ok := queriesByPanel[panelId]
+	if !ok {
 		return dtos.MetricRequest{}, ErrPublicDashboardPanelNotFound
 	}
 
 	ts := publicDashboard.BuildTimeSettings(dashboard)
```

Comment:
nit: slight reorg of blocks in this method and a little documentation

![image](https://user-images.githubusercontent.com/1669620/186549885-3bdce7a1-9b03-4d3b-a099-7e3252f4a154.png)